pennylane/math/quantum.py

```diff
@@ -956,6 +957,10 @@ def sqrt_matrix(density_matrix):
     Returns:
         (tensor_like): Square root of the density matrix.
     """
+    sparse_type = None
+    if issparse(density_matrix):
+        sparse_type = type(density_matrix)
+        density_matrix = density_matrix.toarray()

```

Comment:
Do we want to do this? If the DM is sparse won't this densify it?
Can't we just use https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csr_matrix.sqrt.html on the given sub-terms, or other equivalents in the scipy.sparse module?

Also, not sure if sparse mats support broadcasting/batching

Reply:
To get the matrix sqrt, as far as my knowledge, another practical method is to make use of Schur decomp. `scipy.linalg` has it, but again `scipy.sparse.linalg` do not have.

Reply:
Ah, my bad, I assumed that was the mat sqrt. It seems the sparse linalg routines for `power` also only accept integers. I think this is something then we'd need to implement for support, otherwise we are limiting ourselves in the overall mat size we can represent

Reply:
On the $\text{sqrtmatrix}(Id - A^\dagger A)$, I believe we can temporarily employ the [Denman–Beavers iterations](https://en.wikipedia.org/wiki/Square_root_of_a_matrix#By_Denman%E2%80%93Beavers_iteration) to ensure only sparse operations were applied and meanwhile get the correct results for most of random matrices. But its performance is notably poor for `scipy.sparse`, and essentially it does not naturally preserve the sparsity of matrices at all.

Reply:
Absorbed into on-going story sc-84693